### PR TITLE
fix: make projects page scrollable

### LINKED ISSUE #3694: [ENHANCEMENT] Scrollable Project Page
**Is your feature request related to a problem? Please describe.**
If there are more than 16 projects in database, the UI can not fully display them all, and the page is not scrllable.
image

**Describe the solution you'd like**

More friendly UI for the project page.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by axiomofjoy**
resolves #3694

### CODE REVIEW

**Comment by mikeldking** [comment]
Screenshot 2024-07-01 at 10 43 03 AM
There's a permenant scrollbar on the bottom of the page causing some padding issues

**Comment by mikeldking** [comment]
I'm not sure I'd do scroll based pagination here - but I may be wrong. LMK if you have something I could take a look at. Would like to try out some edge-cases before we ship.

File: app/src/pages/projects/ProjectsPage.tsx [hunk 1]
```diff
@@ -125,7 +161,16 @@ export function ProjectsPageContent({ timeRange }: { timeRange: TimeRange }) {
   );
 
   return (
-    <Flex direction="column" flex="1 1 auto">
+    <div
+      css={css`
+        flex: 1 1 auto;
+        flex-direction: column;
+        overflow: auto;
+        padding-bottom: var(--ac-global-dimension-size-750);
```
Comment (mikeldking) [inline app/src/pages/projects/ProjectsPage.tsx:169]:
```suggestion
        overflow-y: auto;
        overflow-x: hidden;
```

File: app/src/pages/projects/ProjectsPage.tsx [hunk 2]
```diff
@@ -125,7 +161,16 @@ export function ProjectsPageContent({ timeRange }: { timeRange: TimeRange }) {
   );
 
   return (
-    <Flex direction="column" flex="1 1 auto">
+    <div
+      css={css`
+        flex: 1 1 auto;
+        flex-direction: column;
```
Comment (mikeldking) [inline app/src/pages/projects/ProjectsPage.tsx:167]: is this needed?

File: app/src/pages/projects/ProjectsPage.tsx (hunk 1 above)
Comment (axiomofjoy) [reply app/src/pages/projects/ProjectsPage.tsx:183]: addressed in c88eab7d33182a95cc847bbd06b31e3a4cd9f1bc

File: app/src/pages/projects/ProjectsPage.tsx (hunk 2 above)
Comment (axiomofjoy) [reply app/src/pages/projects/ProjectsPage.tsx]: addressed in 122de8e8129e41353cfc203bd03317f38e6b1ed6

File: app/src/pages/projects/ProjectsPage.tsx [hunk 3]
```diff
@@ -90,6 +108,24 @@ export function ProjectsPageContent({ timeRange }: { timeRange: TimeRange }) {
   );
   const projects = projectsData.projects.edges.map((p) => p.project);
 
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const fetchMoreOnBottomReached = useCallback(
+    (containerRefElement?: HTMLDivElement | null) => {
+      if (containerRefElement) {
+        const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
+        //once the user has scrolled within 300px of the bottom of the table, fetch more data if there is any
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply app/src/pages/projects/ProjectsPage.tsx:116]: addressed in 4025521993e870a1a43a975e56ec4fdbb582c6e6

File: app/src/pages/projects/ProjectsPage.tsx [hunk 4]
```diff
@@ -90,6 +108,24 @@ export function ProjectsPageContent({ timeRange }: { timeRange: TimeRange }) {
   );
   const projects = projectsData.projects.edges.map((p) => p.project);
 
+  const tableContainerRef = useRef<HTMLDivElement>(null);
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply app/src/pages/projects/ProjectsPage.tsx:111]: addressed in 4025521993e870a1a43a975e56ec4fdbb582c6e6

**Review by mikeldking — APPROVED** [review]

**Pull request merged**